pycqed/measurement/sweep_points.py

```diff
@@ -4,30 +4,77 @@
 
 
 class SweepPoints(list):
-    def __init__(self, param_name=None, values=None, unit='', param_label=None):
+    """
+    This class is used to create sweep points for any measurement.
+    The SweepPoints object is a list of dictionaries of the form:
+        [
+            # 1st sweep dimension
+            {param_name0: (values, unit, label),
+             param_name1: (values, unit, label),
+            ...
+             param_nameN: (values, unit, label)},
+
+            # 2nd sweep dimension
+            {param_name0: (values, unit, label),
+             param_name1: (values, unit, label),
+            ...
+             param_nameN: (values, unit, label)},
+
+             .
+             .
+             .
+
+            # D-th sweep dimension
+            {param_name0: (values, unit, label),
+             param_name1: (values, unit, label),
+            ...
+             param_nameN: (values, unit, label)},
+        ]
+
+    Example how to use this class to create a 2D sweep for 3 qubits, where
+    the first (hard) sweep is over amplitudes and the :
+
+    sp = SweepPoints()
+    for qb in ['qb1', 'qb2', 'qb3']:
+        sp.add_sweep_parameter(f'amps_{qb}', np.arange(100),
+        'V', 'Pulse amplitude')
+    sp.add_sweep_dimension()
+    for qb in ['qb1', 'qb2', 'qb3']:
+        sp.add_sweep_parameter(f'lengths_{qb}', np.arange(100),
+        'V', 'Pulse length')
+    """
```

Comment:
Useful docstring!

However an example that sweeps an amplitude to 99V and pulse length to 99 V! might be a bit confusing.

Reply:
Thanks :)